pkg/serviceaccount/tokens_controller.go

```diff
@@ -270,6 +282,21 @@ func (e *TokensController) createSecretIfNeeded(serviceAccount *api.ServiceAccou
 
 // createSecret creates a secret of type ServiceAccountToken for the given ServiceAccount
 func (e *TokensController) createSecret(serviceAccount *api.ServiceAccount) error {
+	// We don't want to update the cache's copy of the service account
+	// so add the secret to a freshly retrieved copy of the service account
```

Comment:
This was serving two purposes:
1. Getting from the apiserver gets us a copy we can mutate and update
2. More importantly, we also get to skip a create and cleanup if the event we're processing is outdated. If multiple things are adding secrets to service accounts, several update events can queue up in the process queue. Not doing this check results in a lot of thrash as different controllers attempt to respond to serviceaccount updates by ensuring their particular type of secret is generated, then added, only to get a resource conflict and have to clean up their secret and try again.